apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/backend/ast/AstBuilder.kt

```diff
@@ -665,18 +643,14 @@ internal class AstBuilder private constructor(
           "BOOLEAN" -> CodeGenerationAst.FieldType.Scalar.Boolean(nullable = true)
           "FLOAT" -> CodeGenerationAst.FieldType.Scalar.Float(nullable = true)
           else -> {
-            val customType = checkNotNull(customScalarsMapping[schemaTypeRef.name]) {
+            val className = checkNotNull(customScalarsMapping[schemaTypeRef.name]) {
               "Failed to resolve custom scalar type `${schemaTypeRef.name}`"
             }
             CodeGenerationAst.FieldType.Scalar.Custom(
                 nullable = true,
                 schemaType = schemaTypeRef.name,
-                type = customType,
-                customEnumType = CodeGenerationAst.TypeRef(
-                    name = schemaTypeRef.name,
-                    packageName = typesPackageName,
-                    enclosingType = CodeGenerationAst.customTypeRef(typesPackageName)
-                )
+                type = className,
+                typeName = ClassName(typesPackageName, "CustomScalars", schemaTypeRef.name.capitalize())
```

Comment:
Should we introduce `object CustomScalarType` in `CustomScalarType.kt` with the const for enum name `CustomScalars`. Also I was thinking maybe we should use `object`  as namespace for any top level functions?  For example move all functions in `ObjectTypeCodegen.kt` into `object ObjectTypeCodegen `? So any top level functions will have explicit namespaces

Reply:
> Should we introduce object CustomScalarType in CustomScalarType.kt with the const for enum name CustomScalars

Not sure I get it. `CustomScalars` is now a generated `object`. I'd rather avoid having 2 instances in 2 different packages. 

> Also I was thinking maybe we should use object as namespace for any top level functions?

Yea, it's a bit heterogeneous at the moment. I don't mind it too much since this is mostly internal `compiler` API. The functions in `ObjectType.kt` are mostly extension functions though so IntelliJ will auto-add any enclosing `object` and even if we wrap them, they will look the same at the call site, only the import will change:

```
// this import is currently not needed because everything is in the same package.
import com.apollographql.apollo.compiler.backend.codegen.ObjectTypeCodegen

... 

dataType.typeSpec()
```

It's still convenient being able to completely leave out the import?